src/thunderbots/software/ai/hl/stp/play/play.cpp

```diff
@@ -29,17 +29,21 @@ std::optional<std::vector<std::shared_ptr<Tactic>>> Play::getTactics(const World
     return std::nullopt;
 }
 
-std::vector<std::shared_ptr<Tactic>> Play::getNextTacticsWrapper(
-    TacticCoroutine::push_type &yield)
+void Play::getNextTacticsWrapper(TacticCoroutine::push_type &yield)
 {
     // Yield an empty vector the very first time the function is called. This value will
     // never be seen/used by the rest of the system.
     yield({});
 
     // Anytime after the first function call, the getNextTactics function will be
-    // used to perform the real logic.
+    // used to perform the real logic. The calculateNextIntent function will yield its
+    // values to the top of the coroutine stack, where they will be retrieved by
+    // getNextIntent, so we do not need to yield or return the result of this function
+    //
     // The getNextTactics function is given the World as a parameter rather than using
     // the member variable since it's more explicit and obvious where the World
-    // comes from when implementing Plays
-    return getNextTactics(yield, this->world);
+    // comes from when implementing Plays. The World is passed as a reference, so when
+    // the world member variable is updated the implemented Plays will have access
+    // to the updated world as well.
+    getNextTactics(yield, this->world);
```

Comment:
I think we should just reference the `world` member variable in the play implementation rather then pass it in here. This keeps the Plays inline with how Tactics and Actions get state updates.

Reply:
fixed